internal/process/deprovisioning/edp_deregistration.go

```diff
@@ -98,16 +103,13 @@ func (s *EDPDeregistrationStep) Run(operation internal.Operation, log *slog.Logg
 
 func (s *EDPDeregistrationStep) handleError(operation internal.Operation, err error, log *slog.Logger, msg string) (internal.Operation, time.Duration, error) {
 	if kebError.IsTemporaryError(err) {
-		since := time.Since(operation.UpdatedAt)
-		if since < time.Minute*30 {
-			log.Warn(fmt.Sprintf("request to EDP failed: %s. Retry...", err))
-			return operation, 10 * time.Second, nil
-		}
+		return s.operationManager.RetryOperationWithoutFail(operation, s.Name(), "request to EDP failed", edpRetryInterval, edpRetryTimeout, log, err)
```

Comment:
As I understand it if we returned `return operation, 10 * time.Second, nil` in previous implementation in that if statement and the step would timeout using operation timeout mechanism then eventually the operation would be finished in failed state (see staged_manager.go - Execute method)? If my understanding is correct then we should use `RetryOperation` here.

Reply:
as I remember, there was an agreement that we should only fail if k8s cluster still exists, in other cases - use "executedNotCompleted" logic

Reply:
Good point. So in this step we should not fail.